relay-server/src/actors/processor.rs

```diff
@@ -3343,8 +3322,51 @@ mod tests {
         if let Trace(context) = trace_context {
             assert!(context.sampled.value().unwrap())
         }
+    }
+
+    #[tokio::test]
+    async fn test_error_is_not_tagged_if_inputs_are_invalid() {
```

Comment:
Should such tests rather be unit tests on one of the util functions? It's great to have a few tests covering the full call hierarchy, though the differences between these tests actually occur within `is_trace_fully_sampled`, if I read this correctly.

Reply:
Yep, I didn't refactor them since i wanted to wait for your ok first